docs/deployment/resource-providers/native_kubernetes.md

```diff
@@ -24,394 +23,244 @@ specific language governing permissions and limitations
+
+- **ClusterIP**: Exposes the service on a cluster-internal IP.
+  The Service is only reachable within the cluster.
+  If you want to access the Job Manager ui or submit job to the existing session, you need to start a local proxy.
+  You can then use `localhost:8081` to submit a Flink job to the session or view the dashboard.
 
-Use the following command to start a Flink application.
 {% highlight bash %}
-$ ./bin/flink run-application -p 8 -t kubernetes-application \
-  -Dkubernetes.cluster-id=<ClusterId> \
-  -Dtaskmanager.memory.process.size=4096m \
-  -Dkubernetes.taskmanager.cpu=2 \
-  -Dtaskmanager.numberOfTaskSlots=4 \
-  -Dkubernetes.container.image=<CustomImageName> \
-  local:///opt/flink/usrlib/my-flink-job.jar
+$ kubectl port-forward service/<ServiceName> 8081
 {% endhighlight %}
-</div>
 
-<div data-lang="python" markdown="1">
-{% highlight dockerfile %}
-FROM flink
+- **NodePort**: Exposes the service on each Node’s IP at a static port (the `NodePort`).
+  `<NodeIP>:<NodePort>` could be used to contact the Job Manager Service.
+  `NodeIP` could be easily replaced with Kubernetes ApiServer address.
+  You could find it in your kube config file.
 
-# install python3 and pip3
-RUN apt-get update -y && \
-    apt-get install -y python3.7 python3-pip python3.7-dev && rm -rf /var/lib/apt/lists/*
-RUN ln -s /usr/bin/python3 /usr/bin/python
+- **LoadBalancer**: Exposes the service externally using a cloud provider’s load balancer.
+  Since the cloud provider and Kubernetes needs some time to prepare the load balancer, you may get a `NodePort` JobManager Web Interface in the client log.
+  You can use `kubectl get services/<cluster-id>-rest` to get EXTERNAL-IP and then construct the load balancer JobManager Web Interface manually `http://<EXTERNAL-IP>:8081`.
 
-# install Python Flink
-RUN pip3 install apache-flink
-COPY /path/of/python/codes /opt/python_codes
+Please reference the official documentation on [publishing services in Kubernetes](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types) for more information.
 
-# if there are third party python dependencies, users can install them when building the image
-COPY /path/to/requirements.txt /opt/requirements.txt
-RUN pip3 install -r requirements.txt
+### Accessing the Logs
 
-# if the job requires external java dependencies, they should be built into the image as well
-RUN mkdir -p $FLINK_HOME/usrlib
-COPY /path/of/external/jar/dependencies $FLINK_HOME/usrlib/
-{% endhighlight %}
+By default, the JobManager and TaskManager will output the logs to the console and `/opt/flink/log` in each pod simultaneously.
+The `STDOUT` and `STDERR` output will only be redirected to the console.
+You can access them via
 
-Use the following command to start a PyFlink application, assuming the application image name is **my-pyflink-app:latest**.
 {% highlight bash %}
-$ ./bin/flink run-application -p 8 -t kubernetes-application \
```

Comment:
I think we need a page for describing in general how to deploy a Python program. What I am not so sure is whether we need this for every resource provider. Here, for example, there is not much specific to K8s. One needs to bundle the Python dependencies and then call `flink` with some special parameters which, hopefully, are the same for all different types of deployments.

Reply:
Maybe @dianfu @shuiqiangchen could share more information here.

Reply:
Thank you for your clarification. At my point of view, it would be better to add a `Deployment` page under Python API for describing how to deploy a Python program on different cluster(standalone/Yarn/K8s).

Reply:
Is the Python deployment specific to the used resource-provider (Yarn, K8s, standalone)? Where exactly in the current structure would you put it @shuiqiangchen? Can you provide the full path (from a navigational point of view, e.g. Deployment -> Resource Providers -> Standalone -> Docker).

Reply:
No, the Python deployment does not require any specified resource-provider. So here I'm in favour of your suggestion that adding a new page to generally describe different deployments. And the full path might be Application Development -> Python API -> Deployment, how do you think about it?

Reply:
Hmm, deployment under development might be a bit confusing. Maybe we can add a `Deployment -> Python` page and link this page from `Application Development -> Python API`?

Reply:
Yes, I feel the same that we should illustrate development and deployment separately. I will add the `Deployment -> Python` page in a new PR.